File: clientC++/clientIce.cpp
#include <Ice/Ice.h>
#include <vector>
#include "ServeurIceMP3.h"

using namespace std;
using namespace serveur;

int main(int argc, char* argv[])
{
	int status = 0;
	Ice::CommunicatorPtr ic;
	try {
		ic = Ice::initialize(argc, argv);
		Ice::ObjectPrx base = ic->stringToProxy("SimplePrinter:default -h 178.62.148.22 -p 10000");
		ServeurIceMP3Prx ServeurIceMP3 = ServeurIceMP3Prx::checkedCast(base);
		if (!ServeurIceMP3)
			throw "Invalid proxy";

		string titre, artiste, chemin, nom;
		vector<string> artistes;
		vector<string> titres;


		int choix = 0;
		while(choix !=8 )
		{
			cout << "1 Pour Ajouter" <<endl;
			cout << "2 Pour Supprimer" <<endl;
			cout << "3 Pour Rechercher un titre" <<endl;
			cout << "4 Pour Rechercher par titre" <<endl;
			cout << "5 Pour Rechercher par auteur" <<endl;
			cout << "6 Pour Lire un fichier" <<endl;
			cout << "7 Pour Arreter la lecture" <<endl;
			cout << "8 Pour Quitter" <<endl;
			cout << "Entrez votre choix " << endl;
			cin >> choix;

			if (choix == 1) //ajout
			{
				cout << "Entrez le titre :" << endl;
				cin >> titre;
				cout << "Entrez l'artiste :" << endl;
				cin >> artiste;
				cout << "Entrez le chemin :" << endl;
				cin >> chemin;
				ServeurIceMP3->ajoutfichier(titre, artiste, chemin);
			}

			else if (choix == 2)//supprimer
			{
				cout << "Entrez le titre :" << endl;
				cin >> titre;
				cout << "Entrez l'artiste :" << endl;
				cin >> artiste;
				ServeurIceMP3->suppression(titre, artiste);
			}

			else if (choix == 3)// recherche un titre
			{
				cout << "Entrez le titre :" << endl;
				cin >> titre;
				cout << "Entrez l'artiste :" << endl;
				cin >> artiste;
				if (ServeurIceMP3->recherche(titre, artiste) == "")
					cout << "Fichier inexistant" <<endl;
				else
					cout << "Fichier trouve" << endl;
			}

			else if (choix == 4)// recherche par titre
			{
				cout << "Entrez le titre :" << endl;
				cin >> titre;
				titres = ServeurIceMP3->rechercheTitre(titre);
				if (titres.empty())
					cout << "Aucun fichier" <<endl;
				else
				{
					for (vector<string>::iterator it = titres.begin(); it!= titres.end(); ++it)
					{
						cout << *it << endl;
					}
				}
			}

			else if (choix == 5)// recherche par auteur
			{
				cout << "Entrez l'artiste :" << endl;
				cin >> artiste;
				artistes = ServeurIceMP3->rechercheAuteur(artiste);
				if (artistes.empty())
					cout << "Aucun fichier" <<endl;
				else
				{
					for (vector<string>::iterator it = artistes.begin(); it!= artistes.end(); ++it)
					{
						cout << *it << endl;
					}
				}
			}

			else if (choix == 6) //lire
			{
				cout << "Lecture de fichier" <<endl;
				cout << "Entrez le titre :" << endl;
				cin >> titre;
				cout << "Entrez l'artiste :" << endl;
				cin >> artiste;
				nom = ServeurIceMP3->lireMp3(titre, artiste);
				if (nom == "")
					cout << "Impossible de lire" <<endl;
				else
				{
					cout << "Debut Lecture a cet addresse" << endl;
					cout << "178.62.148.22:8090/" << nom << endl;
				}
			}

			else if (choix == 7) // arret
			{
				cout << "Stop" <<endl;
				if (ServeurIceMP3->stopMp3(nom) == false)
					cout << "La Musique n'est pas en cours de lecture" <<endl;
				else
					cout << "Arret de la musique" << endl;
			}

		}

		//printer->printString("Hello World!");
	} catch (const Ice::Exception& ex) {
		cerr << ex << endl;
		status = 1;
	} catch (const char* msg) {
		cerr << msg << endl;
		status = 1;
	}
	if (ic)
		ic->destroy();
	return status;
}

File: clientJava/clientIce.java
import java.util.Scanner;
import java.util.Vector;
import java.util.*;

public class clientIce {
	public static void main(String[] args)
	{
		int status = 0;
		Ice.Communicator ic = null;
		try {
			ic = Ice.Util.initialize(args);
			Ice.ObjectPrx base = ic.stringToProxy("SimplePrinter:default -h 178.62.148.22 -p 10000");
			serveur.ServeurIceMP3Prx ServeurIceMP3 = serveur.ServeurIceMP3PrxHelper.checkedCast(base);
			if (ServeurIceMP3 == null) throw new Error("Invalid proxy");

			Scanner sc = new Scanner(System.in);
			int choix = 0;
			String nom= "";
			String[] artistes;
			String[] titres;
			while(choix != 8)
			{
				System.out.println("1 Pour Ajouter");
				System.out.println("2 Pour Supprimer");
				System.out.println("3 Pour Rechercher un titre");
				System.out.println("4 Pour Rechercher par titre");
				System.out.println("5 Pour Rechercher par auteur");
				System.out.println("6 Pour Lire un fichier");
				System.out.println("7 Pour Arreter la lecture");
				System.out.println("8 Pour Quitter");
				System.out.println("Entrez votre choix: ");
				choix = sc.nextInt();
				sc.nextLine();
				if (choix == 1) //ajout
				{
					System.out.println("Fonctionnalite ajout");

					System.out.println("Entrez le Titre: ");
					String titre = sc.nextLine();
					sc.nextLine();
					System.out.println("Entrez l'Artiste: ");
					String artiste = sc.nextLine();
					sc.nextLine();
					System.out.println("Entrez le fichier: ");
					String fichier = sc.nextLine();
					sc.nextLine();
					ServeurIceMP3.ajoutfichier(titre,artiste,fichier);

				}
				else if (choix == 2) //Supprimer
				{
					System.out.println("Fonctionnalite suppression");

					System.out.println("Entrez le Titre: ");
					String titre = sc.nextLine();
					sc.nextLine();
					System.out.println("Entrez l'Artiste: ");
					String artiste = sc.nextLine();
					sc.nextLine();
					ServeurIceMP3.suppression(titre,artiste);

				}
				else if (choix == 3) //recherche un titre
				{
					System.out.println("Fonctionnalite recherche");

					System.out.println("Entrez le Titre: ");
					String titre = sc.nextLine();
					sc.nextLine();
					System.out.println("Entrez l'Artiste: ");
					String artiste = sc.nextLine();
					sc.nextLine();
					if (ServeurIceMP3.recherche(titre,artiste).equals("") == true)
						System.out.println("Fichier inexistant");
					else
						System.out.println("Fichier trouvé");

				}
				else if (choix == 4)// recherche par titre
				{
					System.out.println("Entrez le Titre: ");
					String titre = sc.nextLine();
					sc.nextLine();
					titres = ServeurIceMP3.rechercheTitre(titre);
					if (titres.length == 0)
						System.out.println("Fichier inexistant");
					else
					{
						int cnt = 0;
						while(cnt <= titres.length - 1)
						{
							System.out.println(titres[cnt]);
							cnt++;
						}
					}
				}
				else if (choix == 5)// recherche par auteur
				{
					System.out.println("Entrez l'artiste :");
					String artiste = sc.nextLine();
					sc.nextLine();
					artistes = ServeurIceMP3.rechercheAuteur(artiste);
					if (artistes.length == 0)
						System.out.println("Fichier inexistant");
					else
					{
						int cnt = 0;
						while(cnt <= artistes.length - 1)
						{
							System.out.println(artistes[cnt]);
							cnt++;
						}
					}
				}
				else if (choix == 6) //lire
				{
					System.out.println("Lecture de fichier");
					System.out.println("Entrez le Titre: ");
					String titre = sc.nextLine();
					sc.nextLine();
					System.out.println("Entrez l'Artiste: ");
					String artiste = sc.nextLine();
					sc.nextLine();
					nom = ServeurIceMP3.lireMp3(titre, artiste);
					if (ServeurIceMP3.recherche(titre,artiste).equals("") == true)
						System.out.println("Impossible de lire");
					else
					{
						System.out.println("Debut Lecture a cette addresse");
						System.out.println("178.62.148.22:8090/" + nom );
					}
				}
				else if (choix == 7) // arret
				{
					System.out.println("STOP");
					if (ServeurIceMP3.stopMp3(nom) == false)
						System.out.println("La Musique n'est pas en cours de lecture");
					else
						System.out.println("Arret de la musique");
				}
			}
		} catch (Ice.LocalException e) {
			e.printStackTrace();
			status = 1;
		} catch (Exception e) {
			System.err.println(e.getMessage());
			status = 1;
		}
		if (ic != null) {
			try {
				ic.destroy();
			} catch (Exception e) {
				System.err.println(e.getMessage());
				status = 1;
			}
		}
		System.exit(status);		
	}							
}
